src/main/webapp/js/components/user-managment/roles.jsx

```diff
@@ -1,14 +1,75 @@
-import React from 'react';
-
-const Roles = () => (
-  <div className="container-fluid">
-    <h1>
-      Roles Roles Roles Roles
-      <br />
-      Placeholder Placeholder Placeholder Placeholder
-      <br />
-    </h1>
-  </div>
-);
-
-export default Roles;
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import EntityTable from '../entities/entity-table';
+import { ROLE_ENTITY, PERMISSION_ENTITY } from '../../utils/entity-types';
+
+export const defaultRoleConfig = [{
+  base: false,
+  editable: true,
+  filterable: true,
+  required: false,
+  hidden: false,
+  fieldType: null,
+  entity: null,
+  displayName: 'ID',
+  name: 'id',
+  relatedEntity: PERMISSION_ENTITY,
+  relatedField: null,
+  format: '',
+}, {
+  base: false,
+  editable: true,
+  filterable: true,
+  required: false,
+  hidden: false,
+  fieldType: null,
+  entity: null,
+  displayName: 'Name',
+  name: 'name',
+  relatedEntity: PERMISSION_ENTITY,
+  relatedField: null,
+  format: '',
+}, {
+  base: false,
+  editable: true,
+  filterable: true,
+  required: false,
+  hidden: false,
+  fieldType: null,
+  entity: null,
+  displayName: 'Permissions',
+  name: 'user_role_permissions',
+  relatedEntity: PERMISSION_ENTITY,
+  relatedField: null,
+  format: '',
+},
+];
+
+class Roles extends Component {
+  componentDidMount() {
```

Comment:
you need to fetch the user roles and permissions here, you can use 'fetchEntity' for this